### ISSUE TITLE
Serve errors with TypeScript 2.0.2

### ISSUE THREAD

**Issue by ValeryVS** [angular/angular-cli#1917]
**OS**

`Ubuntu 14.04`

**Versions**

```
angular-cli: 1.0.0-beta.11-webpack.8
node: 6.5.0
os: linux x64
```

**Repro steps.**

Create fresh project with webpack.8 or update existed.
There will be `"typescript": "^2.0.0"` dependency in `package.json`.

The log given by the failure.

```
ERROR in [default] 
Cannot find global type 'Array'.

ERROR in [default] 
Cannot find global type 'Boolean'.

ERROR in [default] 
Cannot find global type 'Function'.

ERROR in [default] 
Cannot find global type 'IArguments'.

ERROR in [default] 
Cannot find global type 'Number'.

ERROR in [default] 
Cannot find global type 'Object'.

ERROR in [default] 
Cannot find global type 'RegExp'.

ERROR in [default] 
Cannot find global type 'String'.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/@angular/common/src/directives/ng_class.d.ts:81:33 
Cannot find name 'Set'.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/@angular/common/src/directives/ng_if.d.ts:37:76 
Cannot find name 'Object'.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/@angular/common/src/directives/ng_plural.d.ts:78:53 
Cannot find name 'Object'.

...

Multiple "Cannot find name" errors in @angular and rjxs files here.

...

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:17:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.core.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:18:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.collection.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:19:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.generator.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:20:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.iterable.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:21:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.promise.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:22:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.proxy.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:23:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.reflect.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:24:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.symbol.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:25:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts' not found.

ERROR in [default] /home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es2015.d.ts:26:0 
File '/home/valeryvs/Git/angular-cli-test-3/node_modules/typescript/lib/lib.es5.d.ts' not found.
```

**Workaround**

Lock TypeScript version at 2.0.0 in `package.json`.

```
-"typescript": "^2.0.0"
+"typescript": "2.0.0"
```

**Comment by eyedol** [comment]
@ValeryVS see [#1901](https://github.com/angular/angular-cli/issues/1901#issuecomment-243679698)

**Comment by ValeryVS** [comment]
Right, duplicate. Missed that issue.

**Comment by rakibler** [comment]
Sorry, mine sort of had a bad title

**Comment by dematic-rodrigo-silveira** [comment]
Having the same problem, See [issue 1887](https://github.com/angular/angular-cli/issues/1887) for more details.

**Comment by SeekingMonkey** [comment]
Check out dematic-rodrigo-silveira's post on https://github.com/angular/angular-cli/issues/1919.  I will summarize below.  Huge thanks to dematic-rodrigo-silveira!

For a new project, there are a few steps:

Update Node to 6.5 and npm to 3.10.3
- 	You can check your versions with `node -v` and `npm -v` respectively.
- 	These can be both be updated from the nodejs.org download site.  The installer installs both of them.  https://nodejs.org/en/download/current/

Update Typescript to 2.0+
- 	check with `tsc -v`
- 	`npm install typescript@next` will update you to 2.1.0

Follow the README's instructions on installing the new cli
- `npm uninstall -g angular-cli`
- `npm install -g angular-cli@webpack`
- `npm new projectname` Creates your new project
- `cd projectname` moves you into the newly created directory

Modify project dependencies
- Open package.json from your apps root directory
- Under "devDependancies", locate "typescript".
- Change "^2.0.0" to "2.0.0".  This locks yourproject at typescript 2.0.0 instead of 2.1.0.
- Save

`ng serve`
app works!
`ng init`
`npm install --save-dev angular-cli@webpack` to set the correct cli

Once this is complete, you should be able to serve and test without issues.  I am loving the new cli.  

Thank you to the CLI team!
Happy coding!

**Comment by billdwhite** [comment]
The problem with the solution from @SeekingMonkey is that now, RC6 has been released and it [appears to have a dependency on 2.0.2 specifically ](https://github.com/angular/angular/blob/master/CHANGELOG.md).  What now?

**Comment by aszechlicki** [comment]
we probably need to wait with updating until angular-cli team fixes that
fortunately RC6 isn't as revolutionary as RC5, it's mostly cleanup so waiting a couple days (hopefully) won't change much

**Comment by iBasit** [comment]
+1

**Comment by emyann** [comment]
Very weird issue. Whenever I scaffold a new project with angular-cli: 1.0.0-beta.11-webpack.8 and try to build the project for the first time, I get a list of errors.
My workaround was simply to uninstall / install typescript and the project builds fine: 
[Asciinema record](https://asciinema.org/a/af8qoyxwn6joj1zj34jtk6jy2)

**Comment by CalvinDale** [comment]
I followed the process outlined by @SeekingMonkey. `ng serve` still threw the same list of exceptions. Then I tried @emyann's process and it started working.

**Comment by Blak3Nick** [comment]
I concur with @CalvinDale, I, also, could not get the process @SeekingMonkey described to work, but @emyann's worked like a charm.

**Comment by mcferren** [comment]
me too - Thanks @emyann

**Comment by eosimosu** [comment]
The new cli actually requires typescript@2.0.0

**Comment by filipesilva** [comment]
Closed as obsolete. The latest versions support typescript@2.0.2.

**Comment by yves-s** [comment]
I still get the error as mentioned above:
ERROR in [default]
Property 'Error' does not exist on type 'typeof error'.
ERROR in [default] 
Module 'webdriver.error' has no exported member 'IErrorCode'.
...
ERROR in [default]
Module 'webdriver' has no exported member 'Locator'.

I already tried it with several ts ng combinations like:
tsc: 2.0.0
tsc: 2.0.2

angular-cli: 1.0.0-beta.17
angular-cli: 1.0.0-beta.16
angular-cli: 1.0.0-beta.15

node: 6.5.0

**Comment by filipesilva** [comment]
@yves-s yours seems like a different issue, related to protractor. Can you open a new issue with repro steps?
